packages/x-date-pickers/src/DateField/tests/editing.DateField.test.tsx

```diff
@@ -510,6 +510,65 @@ describe('<DateField /> - Editing', () => {
     });
   });
 
+  describe('Pasting', () => {
+    const firePasteEvent = (input: HTMLInputElement, pastedValue: string) => {
+      act(() => {
+        const clipboardEvent = new Event('paste', {
+          bubbles: true,
+          cancelable: true,
+          composed: true,
+        });
+
+        // @ts-ignore
+        clipboardEvent.clipboardData = {
+          getData: () => pastedValue,
+        };
+        input.dispatchEvent(clipboardEvent);
+      });
+    };
+
+    it('should set the date when all sections are selected the pasted value is valid and a value is provided', () => {
+      const onChange = spy();
+      render(<DateField onChange={onChange} defaultValue={adapterToUse.date()} />);
+      const input = screen.getByRole('textbox');
+      clickOnInput(input, 1);
+
+      // Select all sections
+      userEvent.keyPress(input, { key: 'a', ctrlKey: true });
+
+      firePasteEvent(input, '09/16/2022');
+      expect(onChange.callCount).to.equal(1);
+      expect(onChange.lastCall.firstArg).toEqualDateTime(new Date(2022, 8, 16));
+    });
+
+    it('should set the date when all sections are selected and the pasted value is valid and no value is provided', () => {
+      const onChange = spy();
+      render(<DateField onChange={onChange} />);
+      const input = screen.getByRole('textbox');
+      clickOnInput(input, 1);
+
+      // Select all sections
+      userEvent.keyPress(input, { key: 'a', ctrlKey: true });
+
+      firePasteEvent(input, '09/16/2022');
+      expect(onChange.callCount).to.equal(1);
+      expect(onChange.lastCall.firstArg).toEqualDateTime(new Date(2022, 8, 16));
+    });
+
+    it('should not set the date when all sections are selected and props.readOnly = true', () => {
+      const onChange = spy();
+      render(<DateField onChange={onChange} readOnly />);
+      const input = screen.getByRole('textbox');
+      clickOnInput(input, 1);
+
+      // Select all sections
+      userEvent.keyPress(input, { key: 'a', ctrlKey: true });
+
+      firePasteEvent(input, '09/16/2022');
+      expect(onChange.callCount).to.equal(0);
+    });
```

Comment:
Would be nice to also add a test about what append when the past value is wrong

Reply:
Done :+1: